Modules/Core/Common/include/itkByteSwapper.hxx

```diff
@@ -273,10 +274,8 @@ template <typename T>
 void
 ByteSwapper<T>::Swap2(void * pin)
 {
-  auto *               p = static_cast<unsigned short *>(pin);
-  const unsigned short h1 = (*p) << static_cast<short unsigned int>(8);
-  const unsigned short h2 = (*p) >> static_cast<short unsigned int>(8);
-  *p = h1 | h2;
```

Comment:
Note that the original code here would have undefined behavior if the pointer passed as argument would point to a type _other_ than signed or unsigned `short`.